packages/@react-aria/focus/src/useFocusRing.ts

```diff
@@ -34,10 +34,20 @@ interface FocusRingAria {
  * not with a mouse, touch, or other input methods.
  */
 export function useFocusRing(props: FocusRingProps = {}): FocusRingAria {
-  let {within} = props;
+  let {isTextInput, within} = props;
   let [isFocused, setFocused] = useState(false);
   let [isFocusWithin, setFocusWithin] = useState(false);
-  let {isFocusVisible} = useFocusVisible(props);
+  let [isFocusVisible, setFocusVisible] = useState(false);
```

Comment:
This should default to `autoFocus && isFocusVisible()` in case the component mounts with auto focus and focus is already visible.